src/main/java/io/stargate/sgv2/jsonapi/service/operation/tables/TableReadAttemptBuilder.java

```diff
@@ -61,13 +61,27 @@ public ReadAttempt<TableSchemaObject> build(WhereCQLClause<Select> whereCQLClaus
 
     readPosition += 1;
 
-    return new ReadAttempt<>(
-        readPosition,
-        tableSchemaObject,
-        selectCQLClause,
-        whereCQLClause,
-        cqlOptions,
-        pagingState,
-        documentSourceSupplier);
+    final WhereCQLClauseAnalyzer.WhereCQLClauseAnalyzeResult whereCQLClauseAnalyzeResult =
+        whereCQLClause.analyseWhereClause();
+    // Analyse filter usage
+    // Step 1, may add AllowFiltering
+    if (whereCQLClauseAnalyzeResult.withAllowFiltering()) {
+      cqlOptions.addBuilderOption(CQLOption.ForSelect.withAllowFiltering());
```

Comment:
this is a bug - it will set the allow filtering on the CqlOperations object that is used everytime the build() function is called. 

add a copy consturctor to the CqlOptions that takes a CqlOperations as a param and copies the contents, then create a new instance of CqlOperations in this function if we need to allow filtering and use that for the read attempt.

Reply:
Solved. Thanks